daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/CommandsVersion.scala

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf.engine
+
+import com.digitalasset.daml.lf.transaction.VersionTimeline
+import com.digitalasset.daml.lf.language.{LanguageVersion, LanguageMajorVersion => LMV}
+import com.digitalasset.daml.lf.data.Ref._
+import com.digitalasset.daml.lf.command._
+import com.digitalasset.daml.lf.data.{ImmArray, ImmArrayCons}
+import com.digitalasset.daml.lf.language.Ast.Package
+
+import scala.annotation.tailrec
+
+/** Tx authorization is parametrized by the language version that "originated it".
```

Comment:
the submitter check is not an authorization check.

Suggestion: adapt comment to state that command interpretation can depend on the language version that the command originated from. For example <text about submitter check>.

This way the explanation is more context independent.

Reply:
Done